src/output-tx.c

```diff
@@ -242,10 +241,13 @@ static TmEcode OutputTxLog(ThreadVars *tv, Packet *p, void *thread_data)
             BUG_ON(logger != NULL && store == NULL);
         }
 
-        if (!logger_not_logged) {
-            SCLogDebug("updating log tx_id %"PRIu64, tx_id);
-            AppLayerParserSetTransactionLogId(f->alparser);
-        }
+    }
+
+    /* Update the greatest transaction logged if all loggers
+     * succeeded. */
+    if (!not_logged) {
+        SCLogDebug("updating log tx_id %"PRIu64, tx_id);
+        AppLayerParserSetTransactionLogId(f->alparser, tx_id);
```

Comment:
Since the above loop cannot be broken out of, tx_id will always be equal to total_txs AFAICT. That would mean that if any tx is logged succesfully, we update log_id to the highest?